realm/realm-library/src/main/cpp/io_realm_SyncManager.cpp

```diff
@@ -96,3 +96,23 @@ JNIEXPORT void JNICALL Java_io_realm_SyncManager_nativeConfigureMetaDataSystem(J
     }
     CATCH_STD()
 }
+
+JNIEXPORT void JNICALL Java_io_realm_SyncManager_nativeSimulateSyncError(JNIEnv* env, jclass, jstring localRealmPath,
+                                                                         jint errorCode, jstring errorMessage,
+                                                                         jboolean isFatal)
+{
+    TR_ENTER()
+    try {
+        JStringAccessor local_realm_path(env, localRealmPath);
+        JStringAccessor error_message(env, errorMessage);
+
+        std::shared_ptr<SyncSession> session = SyncManager::shared().get_existing_active_session(local_realm_path);
```

Comment:
I don't mind changing it, but I'm not sure I fully understand why, especially since we later do `*session` which would only work if this was a pointer type?